Demo/Demo/SwiftUIComponents/SwiftUIPaymentButtonDemo.swift

```diff
@@ -1,12 +1,109 @@
+
+    @ViewBuilder
     var body: some View {
         ZStack {
-            Text("WIP - SwiftUI Button Customization View")
+            VStack {
+                Picker("Funding Source", selection: $fundingIndex) {
+                    ForEach(fundingSources.indices) { index in
+                        Text(fundingSources[index])
+                    }
+                }
+                .pickerStyle(SegmentedPickerStyle())
+                .onChange(of: fundingIndex) { _ in
+                    selectedFunding = PaymentButtonFundingSource.allCases()[fundingIndex]
+                    colors = getColorFunding(with: selectedFunding)
+                    colorsIndex = 0
+                    pickerId += 1 // Hack to change ID of picker. ID is updated to force refresh, https://developer.apple.com/forums/thread/127560
+                    buttonId += 1
+                }
+
+                Picker("Colors", selection: $colorsIndex) {
+                    ForEach(colors.indices) { index in
+                        Text(colors[index])
+                    }
+                }
+                .pickerStyle(SegmentedPickerStyle())
+                .onChange(of: colorsIndex) { _ in
+                    buttonId += 1
+                }
+                .id(pickerId)
+
+                Picker("Edges", selection: $edgesIndex) {
+                    ForEach(edges.indices) { index in
+                        Text(edges[index])
+                    }
+                }
+                .pickerStyle(SegmentedPickerStyle())
+                .onChange(of: edgesIndex) { _ in
+                    selectedEdge = PaymentButtonEdges.allCases()[edgesIndex]
+                    buttonId += 1
+                }
+
+                Picker("sizes", selection: $sizesIndex) {
+                    ForEach(sizes.indices) { index in
+                        Text(sizes[index])
+                    }
+                }
+                .pickerStyle(SegmentedPickerStyle())
+                .onChange(of: sizesIndex) { _ in
+                    selectedSize = PaymentButtonSize.allCases()[sizesIndex]
+                    buttonId += 1
+                }
+
+                if selectedFunding == .payPal {
+                    PayPalButton(color: PayPalButton.Color.allCases()[colorsIndex], edges: selectedEdge, size: selectedSize).id(buttonId)
+                } else if selectedFunding == .payLater {
+                    PayPalPayLaterButton(
+                        color: PayPalPayLaterButton.Color.allCases()[colorsIndex],
+                        edges: selectedEdge,
+                        size: selectedSize
+                    ).id(buttonId)
+                } else {
+                    PayPalCreditButton(
+                        color: PayPalCreditButton.Color.allCases()[colorsIndex],
+                        edges: selectedEdge,
+                        size: selectedSize
+                    ).id(buttonId)
+                }
```

Comment:
```suggestion
                switch selectedFunding {
                case .payPal:
                    PayPalButton(color: PayPalButton.Color.allCases()[colorsIndex], edges: selectedEdge, size: selectedSize).id(buttonId)

                case .payLater:
                    PayPalPayLaterButton(
                        color: PayPalPayLaterButton.Color.allCases()[colorsIndex],
                        edges: selectedEdge,
                        size: selectedSize
                    ).id(buttonId)

                case .credit:
                    PayPalCreditButton(
                        color: PayPalCreditButton.Color.allCases()[colorsIndex],
                        edges: selectedEdge,
                        size: selectedSize
                    ).id(buttonId)
                }
```

Reply:
I think we can use a switch here instead to clean this up and make it a bit more swift-y